documentation: added overview (table) of requisites to requisites doc

sypets opened the pull request:
This is an addition to the documentation page on requisites. It adds a table of requisites as a handy overview showing the differences in the behaviour of the requisites, e.g. execute only if target has changes, execute only if target is successful / fails, execution order etc. 

This PR does not handle an open issue.

Push to "develop" because it is more of a new feature than a fix.

Commit:
documentation: added overview (table) of requisites to requisites doc

cachedout commented:
I think this is a great idea!

One thing we might do to make this more clear is to change the second column to indicate the action is based on the `result` field in the return and it can be `True` or `False` (instead of `ok`). Thoughts?

sypets commented:
@cachedout Thanks for your feedback. I think your suggestion is good. If you already have an idea how to phrase the second column, please feel free. 

I tried to make the table as clear and concise as possible, but it is not so easy ...

Commit:
changes for more clarity

whiteinge commented:
Awesome addition!

Two suggestions for prereq.

> State is only executed if target:
> has changes (run individually as dry-run)

> Comment:
> inserts itself as requisite into target state

Commit:
added info for prereq, changed heading text and formatting

sypets commented:
Added another commit. Incorporated the suggestions except comment for prereq, 'cause I'm not sure whether this is relevant for understanding functionality.

whiteinge commented:
I somehow overlooked your 'Order' column entirely. My suggestion for the prereq comment is entirely redundant with that. Good call. 👍
